library/src/main/scala/scala/scalajs/js/JSConverters.scala

```diff
@@ -21,7 +21,7 @@ import scala.scalajs.runtime.genTraversableOnce2jsArray
 sealed abstract class JSConvertersLowPrioImplicits { this: JSConverters.type =>
 
   @inline
-  implicit def JSRichFutureNonThenable[A](f: Future[A]): JSRichFuture[A] =
+  implicit def JSRichFutureNonThenable[A, A1](f: A1)(implicit ev: A1 => Future[A]): JSRichFuture[A] =
     new JSRichFuture[A](f.asInstanceOf[Future[A | Thenable[A]]])
```

Comment:
This completely by-passes `ev`. The cast applies without `ev` kicking in first. At least this should be `ev(f).asInstanceOf[...]`.